engine/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java

```diff
@@ -119,6 +119,28 @@ public List<Endpoint> getTargetEndpoints(String orgId, EventType eventType) {
         return endpoints;
     }
 
+    public List<Endpoint> getTargetEndpointsWithoutUsingBgs(String orgId, EventType eventType) {
+        String query = "SELECT DISTINCT e FROM Endpoint e JOIN e.eventTypes " +
+            "WHERE (e.orgId = :orgId OR e.orgId IS NULL) AND e.enabled IS TRUE AND e.status = :status AND :eventType member of e.eventTypes ";
```

Comment:
Did you check which SQL query Hibernate creates from that statement?

Reply:
Yes, generated SQL request is:
`    select
        distinct e1_0.id,
        e1_0.account_id,
        e1_0.endpoint_sub_type,
        e1_0.endpoint_type_v2,
        e1_0.created,
        e1_0.description,
        e1_0.enabled,
        e1_0.name,
        e1_0.org_id,
        e1_0.server_errors,
        e1_0.server_errors_since,
        e1_0.status,
        e1_0.updated 
    from
        endpoints e1_0 
    join
        endpoint_event_type et1_0 
            on e1_0.id=et1_0.endpoint_id 
    where
        (
            e1_0.org_id=? 
            or e1_0.org_id is null
        ) 
        and e1_0.enabled is true 
        and e1_0.status=? 
        and ? in (select
            et2_0.event_type_id 
        from
            endpoint_event_type et2_0 
        where
            e1_0.id=et2_0.endpoint_id)`